tools/run_tests/run_xds_tests.py

```diff
@@ -357,6 +408,62 @@ def wait_until_all_rpcs_go_to_given_backends(backends,
                                    allow_failures=False)
 
 
+def wait_until_rpcs_in_flight(rpc_type, timeout_sec, num_rpcs, threshold):
+    '''Block until the test client reaches the state with the given number
+    of RPCs being outstanding stably.
+
+    Args:
+      rpc_type: A string indicating the RPC method to check for. Either
+        'UnaryCall' or 'EmptyCall'.
+      timeout_sec: Maximum number of seconds to wait until the desired state
+        is reached.
+      num_rpcs: Expected number of RPCs to be in-flight.
+      threshold: Number within [0,100], the tolerable percentage by which
+        the actual number of RPCs in-flight can differ from the expected number.
+    '''
+    if threshold < 0 or threshold > 100:
+        raise ValueError('Value error: Threshold should be between 0 to 100')
+    threshold_fraction = threshold / 100.0
+    start_time = time.time()
+    error_msg = None
+    logger.debug('Waiting for %d sec until %d RPCs (with %d%% tolerance) in-flight'
+                 % (timeout_sec, num_rpcs, threshold))
+    while time.time() - start_time <= timeout_sec:
+        error_msg = _check_rpcs_in_flight(rpc_type,
+                                          num_rpcs,
+                                          threshold,
+                                          threshold_fraction)
+        if error_msg:
+            time.sleep(2)
+        else:
+            break
+    # Ensure the number of outstanding RPCs is stable.
+    if not error_msg:
+        time.sleep(5)
+        error_msg = _check_rpcs_in_flight(rpc_type,
+                                          num_rpcs,
+                                          threshold,
+                                          threshold_fraction)
+    if error_msg:
+        raise Exception(error_msg)
```

Comment:
This error string is just `actual(%d) < expected(%d - %d%%)`. Add detailed context.

Reply:
Enhanced.